hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreCompactor.java

```diff
@@ -74,14 +74,15 @@
 
   int immutCellsNum = 0;  // number of immutable for compaction cells
   /**
-   * Types of Compaction
+   * Types of actions to be done on the pipeline upon MemStoreCompaction invocation
    */
-  private enum Type {
-    COMPACT_TO_SKIPLIST_MAP,
-    COMPACT_TO_ARRAY_MAP
+  private enum Action {
+    COMPACT_DATA_TO_SKIPLIST_MAP,
+    COMPACT_DATA_TO_ARRAY_MAP,
+    COMPACT_INDEX_TO_ARRAY_MAP
   }
 
-  private Type type = Type.COMPACT_TO_ARRAY_MAP;
+  private Action action = Action.COMPACT_DATA_TO_ARRAY_MAP;
```

Comment:
going back -- just wondering why do we need an Action as part of the compactor state -- shouldn't this be determined by the policy (currently a method) why an attribute is needed?
If this is for later distinguishing between cell array and chunk map we can derive this from the compacting memstore.